File: ddd/tests/fixtures/error-contract/workspace/billing-domain/src/naming.rs
//! Declarations whose names collide under a flat name table but not under
//! namespace and scope aware resolution. Each one is reachable as the error type
//! of its own operation, so the symbols they resolve to are observable.

pub enum r#type {
    Raw,
}

pub enum Type {
    Plain,
}

pub mod invoice {
    pub enum invoice {
        Nested,
    }
}

pub struct Invoice;

impl Invoice {
    pub fn raw(&mut self) -> core::result::Result<(), r#type> {
        Ok(())
    }

    pub fn plain(&mut self) -> core::result::Result<(), Type> {
        Ok(())
    }

    pub fn nested(&mut self) -> core::result::Result<(), invoice::invoice> {
        Ok(())
    }
}

File: ddd/tests/fixtures/error-contract/typescript-workspace/billing-domain/src/naming.ts
/**
 * A second declaration that carries the same name as the one in `errors.ts`.
 * Both are reachable as the error type of their own operation, so the symbols
 * they resolve to are observable and must stay apart.
 */

import type { Result } from "./result.ts";

export type IssueInvoiceError = "declared-elsewhere" | "still-elsewhere";

export class Naming {
  issue(): Result<void, IssueInvoiceError> {
    return { ok: true, value: undefined };
  }
}
